### PR TITLE
api doc

### LINKED ISSUE #11: swagger

### PR DESCRIPTION
**Pull request by huhuhu0420**
about 60%~70% of code is generated by copilot, so it may have bugs

### CODE REVIEW

File: backend/api/swagger.yaml [hunk 1]
```diff
@@ -31,166 +32,1077 @@ paths:
...
               schema:
-                $ref: '#/components/schemas/Error'
-  /cart:
-    post:
+                $ref: '#/components/schemas/UserArray'
+        "400":
+          description: bad request
+        "500":
+          description: Unexpected error
+  /admin/get-all-orders:
```
Comment (nonorze-VOXX) [inline backend/api/swagger.yaml:1098]: 同一個訂單的東西放在一起回傳吧
回傳一個2D array

File: backend/api/swagger.yaml [hunk 2]
```diff
@@ -31,166 +32,1077 @@ paths:
...
+              type: string
+              example: "1"
+      requestBody:
+        description: discount id
+        content:
+          application/json:
+            schema:
+              type: integer
+              format: int64
+              example: 1
```
Comment (nonorze-VOXX) [inline backend/api/swagger.yaml]: here look no json in swagger

**Review by nonorze-VOXX — CHANGES_REQUESTED** [review]

File: backend/api/swagger.yaml [hunk 3]
```diff
@@ -31,166 +32,1077 @@ paths:
...
       - email
       - name
-      - phone
       - store_id
+      - rate
+      - rate_count
+      - picture
+      - status
+      - description
       type: object
```
Comment (wj461) [inline backend/api/swagger.yaml]: where is shipping fee

File: backend/api/swagger.yaml [hunk 4]
```diff
@@ -200,34 +1112,41 @@ components:
...
+        $ref: '#/components/schemas/ProductInfo'
+      x-schema-name: ProductArray
     CartInfo:
       required:
       - customer_id
       - product_id
       - product_quantity
-      - store_id
+      - cart_id
+      - product_price
```
Comment (wj461) [inline backend/api/swagger.yaml]: Why cart need product_price
product_id can get price in products table

File: backend/api/swagger.yaml (hunk 1 above)
Comment (huhuhu0420) [reply backend/api/swagger.yaml:1098]: fix

File: backend/api/swagger.yaml (hunk 2 above)
Comment (huhuhu0420) [reply backend/api/swagger.yaml]: i think not important

File: backend/api/swagger.yaml (hunk 3 above)
Comment (huhuhu0420) [reply backend/api/swagger.yaml]: fix

File: backend/api/swagger.yaml (hunk 4 above)
Comment (huhuhu0420) [reply backend/api/swagger.yaml]: fix

File: db/c30-user_data.sql [hunk 5]
```diff
@@ -1,10 +1,12 @@
-CREATE TABLE IF NOT EXISTS user_datas(
+CREATE TABLE IF NOT EXISTS user_data(
```
Comment (nonorze-VOXX) [inline db/c30-user_data.sql:1]: why turn back

**Review by nonorze-VOXX — DISMISSED** [review]
docker-compose up work
LGTM

File: db/c10-store.sql [hunk 6]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE IF NOT EXISTS stores (
+    store_id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    rate FLOAT NOT NULL,
```
Comment (wj461) [inline db/c10-store.sql:4]: the rate is integer in swagger api StoreInfo

File: db/c10-store.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    rate_count INTEGER NOT NULL,
+    address VARCHAR(255) NOT NULL,
+    picture VARCHAR(255),
+    description TEXT,
+    shipping_fee INTEGER NOT NULL,
+    status INTEGER NOT NULL
+);
```
Comment (wj461) [inline db/c10-store.sql:11]: The table no have email
the StoreInfo in swagger have

**Review by wj461 — APPROVED** [review]

File: db/c10-store.sql (hunk 7 above)
Comment (huhuhu0420) [reply db/c10-store.sql:11]: fix

File: db/c30-user_data.sql (hunk 5 above)
Comment (huhuhu0420) [reply db/c30-user_data.sql:1]: no datas

File: backend/api/swagger.yaml (hunk 2 above)
Comment (huhuhu0420) [reply backend/api/swagger.yaml]: fix (maybe)

**Review by nonorze-VOXX — APPROVED** [review]
LGTM

**Pull request merged**